File: src/main/java/io/polygloat/dtos/request/validators/ValidationError.java
package io.polygloat.dtos.request.validators;

import io.polygloat.constants.Message;

public class ValidationError {
    private String[] parameters;

    private ValidationErrorType type;

    private Message message;

    public ValidationError(ValidationErrorType type, Message message, String... parameters) {
        this.parameters = parameters;
        this.message = message;
        this.type = type;
    }

    public String[] getParameters() {
        return this.parameters;
    }

    public ValidationErrorType getType() {
        return this.type;
    }

    public Message getMessage() {
        return this.message;
    }
}

File: backend/data/src/main/kotlin/io/tolgee/dtos/request/validators/ValidationError.kt
package io.tolgee.dtos.request.validators

import io.tolgee.constants.Message

class ValidationError(
  type: ValidationErrorType,
  message: Message,
  vararg parameters: String,
) {
  val parameters: Array<out String> = parameters
  val type: ValidationErrorType = type
  val message: Message = message
}
